### PR TITLE
Detect Cloud Instance by Region as Fallback

### PR DESCRIPTION
**Pull request by AndreasBurger**
**How to categorize this PR?**
/area usability
/kind enhancement
/platform azure

**What this PR does / why we need it**:
With the changes in this PR, the extension will look at the name of the region configured in the shoot to determine the correct cloud instance to connect to iff there is no explicit CloudConfiguration provided.

**Release note**:

```noteworthy user
When no CloudConfiguration is given, the Azure Cloud instance to connect to will be inferred from the region configured for the shoot
```

### CODE REVIEW

File: pkg/apis/azure/helper/helper.go [hunk 1]
```diff
@@ -145,3 +146,24 @@ func InfrastructureZoneToString(zone int32) string {
...
+
+// CloudInstanceNameFromRegion returns the canonical cloud instance name for the instance the region is hosted in.
+func CloudInstanceNameFromRegion(region string) string {
+	if hasAnyPrefix(region, api.AzureGovRegionPrefixes...) {
+		return api.AzureGovCloudName
+	} else if hasAnyPrefix(region, api.AzureChinaRegionPrefixes...) {
+		return api.AzureChinaCloudName
+	} else {
+		return api.AzurePublicCloudName
+	}
```
Comment (kon-angelo) [inline pkg/apis/azure/helper/helper.go:158]: Maybe a switch statement would look cleaner here

Comment (kon-angelo) [reply pkg/apis/azure/helper/helper.go:158]: Also I wouldn't necessarily put this in the api/helper package

File: pkg/controller/backupbucket/actuator.go [hunk 2]
```diff
@@ -41,7 +41,13 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, backupBucket *e
...
 	}
 
-	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(backupConfig.CloudConfiguration)
+	cloudConfiguration := backupConfig.CloudConfiguration
+
+	if cloudConfiguration == nil {
+		cloudConfiguration = &azure.CloudConfiguration{Name: helper.CloudInstanceNameFromRegion(backupBucket.Spec.Region)}
+	}
+
+	azCloudConfiguration, err := azureclient.AzureCloudConfigurationFromCloudConfiguration(cloudConfiguration)
```
Comment (kon-angelo) [inline pkg/controller/backupbucket/actuator.go:50]: Can't these functions be grouped up in a single function ? Can we imagine using one but not the other ?

Comment (AndreasBurger) [reply pkg/controller/backupbucket/actuator.go]: Unfortunately, there is _one_ case where they are both required. However, I agree that duplicating these lines of code tenfold is not the way to go.

I've tried my hand at refactoring it a bit, but unfortunately there is some inherent ugliness here due to the way the upstream cloud-instance structs are defined. Not sure if I have hit the mark, honestly 🤔

**Review by kon-angelo — DISMISSED** [review]
/lgtm

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**